classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php

```diff
@@ -176,11 +223,20 @@ function execute($args, $request) {
 				$this->submission->stampStatusModified();
 				$this->submission->setSubmissionProgress($this->step + 1);
 			}
+			// Add or update comments to editor
+			if ($this->getData('commentsToEditor')){
+				$queries = $queryDao->getByAssoc(ASSOC_TYPE_SUBMISSION, $this->submissionId);
+				$query = $queries->next();
+				if (isset($query)){
+					$this->setCommentsToEditor($this->submissionId, $this->getData('commentsToEditor'), $user->getId(), 1);
```

Comment:
How about giving the `$query` object as the last parameter to the `setCommentsToEditor `function, instead of `$update`? If $query is set than it would be updated and if not a new would be created...